br/pkg/lightning/restore/check_info.go

```diff
@@ -76,42 +76,69 @@ func (rc *Controller) getReplicaCount(ctx context.Context) (uint64, error) {
 }
 
 // ClusterResource check cluster has enough resource to import data. this test can by skipped.
-func (rc *Controller) ClusterResource(ctx context.Context, localSource int64) error {
+func (rc *Controller) ClusterResource(ctx context.Context) error {
 	passed := true
 	message := "Cluster resources are rich for this import task"
 	defer func() {
 		rc.checkTemplate.Collect(Critical, passed, message)
 	}()
 
-	result := &api.StoresInfo{}
-	err := rc.tls.WithHost(rc.cfg.TiDB.PdAddr).GetJSON(ctx, pdStores, result)
+	var (
+		clusterAvail  uint64
+		clusterSource uint64
+	)
+	restoreStarted := false
+	err := rc.taskMgr.CheckTasksExclusively(ctx, func(tx *sql.Tx, tasks []taskMeta) error {
+		clusterAvail = 0
+		clusterSource = 0
+		for _, task := range tasks {
+			if task.status > taskMetaStatusInitial {
+				restoreStarted = true
+				return nil
+			}
+			clusterSource += task.sourceBytes
+			if task.clusterAvail > 0 {
+				clusterAvail = task.clusterAvail
+			}
+		}
+		if clusterAvail > 0 {
+			return nil
+		}
+
+		result := &api.StoresInfo{}
+		if err := rc.tls.WithHost(rc.cfg.TiDB.PdAddr).GetJSON(ctx, pdStores, result); err != nil {
+			return errors.Trace(err)
+		}
+		for _, store := range result.Stores {
+			clusterAvail += uint64(store.Status.Available)
+		}
+		// TODO: Avoid operating meta table outside taskMgr.
+		query := fmt.Sprintf("update %s set cluster_avail = ?",
+			common.UniqueTable(rc.cfg.App.MetaSchemaName, TaskMetaTableName))
+		if _, err := tx.ExecContext(ctx, query, clusterAvail); err != nil {
+			return errors.Trace(err)
+		}
+		return nil
+	})
 	if err != nil {
 		return errors.Trace(err)
 	}
-	totalCapacity := typeutil.ByteSize(0)
-	for _, store := range result.Stores {
-		totalCapacity += store.Status.Capacity
-	}
-	clusterSource := localSource
-	if rc.taskMgr != nil {
-		clusterSource, err = rc.taskMgr.CheckClusterSource(ctx)
-		if err != nil {
-			return errors.Trace(err)
-		}
+	if restoreStarted {
```

Comment:
If store has started we should still check whether the total source is larger than `clusterAvail`, because this task has increase the total source bytes

Reply:
okay, I will always check the total source.